rclcpp/minimal_subscriber/not_composable.cpp

```diff
@@ -15,26 +15,26 @@
 #include "rclcpp/rclcpp.hpp"
 #include "std_msgs/msg/string.hpp"
 
-rclcpp::Node::SharedPtr g_node = nullptr;
-
 /* We do not recommend this style anymore, because composition of multiple
  * nodes in the same executable is not possible. Please see one of the subclass
  * examples for the "new" recommended styles. This example is only included
  * for completeness because it is similar to "classic" standalone ROS nodes. */
 
 void topic_callback(const std_msgs::msg::String::SharedPtr msg)
 {
-  RCLCPP_INFO(g_node->get_logger(), "I heard: '%s'", msg->data.c_str())
+  RCLCPP_INFO(rclcpp::get_logger("minimal_subscriber"), "I heard: '%s'", msg->data.c_str())
```

Comment:
This is not the same thing. This function actually does need the node instance to maintain the existing behavior (and the existing behavior is the desired behavior in my opinion).

Reply:
All right, in that case, I've changed it to manually destroy the subscription first, which achieves the same effect.  I've also updated the comment.  Please take another look when you get a chance.